pkg/api/admin/openshiftcluster_validatestatic.go

```diff
@@ -0,0 +1,290 @@
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeMismatchingResourceID, "id", "The provided resource ID '%s' did not match the name in the Url '%s'.", oc.ID, sv.resourceID)
+	}
+	if !strings.EqualFold(oc.Name, sv.r.ResourceName) {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeMismatchingResourceName, "name", "The provided resource name '%s' did not match the name in the Url '%s'.", oc.Name, sv.r.ResourceName)
+	}
+	if !strings.EqualFold(oc.Type, resourceProviderNamespace+"/"+resourceType) {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeMismatchingResourceType, "type", "The provided resource type '%s' did not match the name in the Url '%s'.", oc.Type, resourceProviderNamespace+"/"+resourceType)
+	}
+	if !strings.EqualFold(oc.Location, sv.location) {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "location", "The provided location '%s' is invalid.", oc.Location)
+	}
+
+	return sv.validateProperties("properties", &oc.Properties)
+}
+
+func (sv *openShiftClusterStaticValidator) validateProperties(path string, p *Properties) error {
+	switch p.ProvisioningState {
+	case ProvisioningStateCreating, ProvisioningStateUpdating, ProvisioningStateAdminUpdating,
+		ProvisioningStateDeleting, ProvisioningStateSucceeded,
+		ProvisioningStateFailed:
+	default:
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".provisioningState", "The provided provisioning state '%s' is invalid.", p.ProvisioningState)
+	}
+	if err := sv.validateClusterProfile(path+".clusterProfile", &p.ClusterProfile); err != nil {
+		return err
+	}
+	if err := sv.validateConsoleProfile(path+".consoleProfile", &p.ConsoleProfile); err != nil {
+		return err
+	}
+	if err := sv.validateServicePrincipalProfile(path+".servicePrincipalProfile", &p.ServicePrincipalProfile); err != nil {
+		return err
+	}
+	if err := sv.validateNetworkProfile(path+".networkProfile", &p.NetworkProfile); err != nil {
+		return err
+	}
+	if err := sv.validateMasterProfile(path+".masterProfile", &p.MasterProfile); err != nil {
+		return err
+	}
+	if len(p.WorkerProfiles) != 1 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".workerProfiles", "There should be exactly one worker profile.")
+	}
+	if err := sv.validateWorkerProfile(path+".workerProfiles['"+p.WorkerProfiles[0].Name+"']", &p.WorkerProfiles[0], &p.MasterProfile); err != nil {
+		return err
+	}
+	if err := sv.validateAPIServerProfile(path+".apiserverProfile", &p.APIServerProfile); err != nil {
+		return err
+	}
+	if len(p.IngressProfiles) != 1 {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".ingressProfiles", "There should be exactly one ingress profile.")
+	}
+	if err := sv.validateIngressProfile(path+".ingressProfiles['"+p.IngressProfiles[0].Name+"']", &p.IngressProfiles[0]); err != nil {
+		return err
+	}
+
```

Comment:
strictly, it seems like validation for Install and StorageSuffix is missing here.  At the moment none of this is too applicable because mutation is disabled.  I think we should discuss what strategy we want to take - part of me thinks that if this validation can't be triggered right now, perhaps it shouldn't be here?